src/zope/index/field/index.py

```diff
@@ -84,8 +76,9 @@ def index_doc(self, docid, value):
     def unindex_doc(self, docid):
         """See interface IInjection"""
         rev_index = self._rev_index
-        value = rev_index.get(docid)
```

Comment:
As far as I know, there's nothing that's actually in the spec about this specifically. 

- Literals can be automatically interned by all implementations.
- Strings can be interned manually.
- PyPy can intern numbers automatically (well, it [fakes that](http://doc.pypy.org/en/latest/cpython_differences.html#object-identity-of-primitive-values-is-and-id))
- CPython uses freelists for things like lists and tuples (and dicts?), so in theory it's possible to observe two such "different" objects having the same ID, *so long as you don't keep a reference to them*. Likewise, PyPy fake-interns tuples.

I think that "keeping a reference" thing is generally enough to be safe, though, so long as you avoid things that can be written literally like strings and numbers (typically one sees `object`). You can keep the `_MARKER` at global module scope or if you want to be extra sure you can cons one up for every function invocation:
```python
marker = object()
value = rev_index.get(docid, marker)
if value is marker:
...
```